alerter.c

```diff
@@ -1,31 +1,77 @@
 #include <stdio.h>
 #include <assert.h>
 
+typedef enum
+{
+    prod,
+    Test
+} Environment;
 int alertFailureCount = 0;
 
-int networkAlertStub(float celcius) {
-    printf("ALERT: Temperature is %.1f celcius.\n", celcius);
-    // Return 200 for ok
-    // Return 500 for not-ok
-    // stub always succeeds and returns 200
-    return 200;
+int networkAlertStub(float celcius) 
+{    
+    if(celcius <= 200)
+    {
+        return 200;
+    }
+    else
+    {
+        return 500;
+    }
 }
-
-void alertInCelcius(float farenheit) {
-    float celcius = (farenheit - 32) * 5 / 9;
-    int returnCode = networkAlertStub(celcius);
-    if (returnCode != 200) {
-        // non-ok response is not an error! Issues happen in life!
-        // let us keep a count of failures to report
-        // However, this code doesn't count failures!
-        // Add a test below to catch this bug. Alter the stub above, if needed.
-        alertFailureCount += 0;
+int networkAlertProd(float celcius)
+{
+    if(celcius <= 200)
+    {
+        return 200;
     }
+    else
+    {
+        return 500;
+    }
+}    
+
+float SelectAlertEnvironment(float celcius, Environment environment_e)
+{
+   float calculated_Value;
+    
+   if(environment_e == prod)
+   {
+     calculated_Value = networkAlertProd(celcius);
+   }
+   else
+   {
+     calculated_Value = networkAlertStub(celcius);
+   }
+   return calculated_Value;
+}
+
+float convertFarenheitToCelcius(float farenheit)
+{
+    float celcius = ((farenheit - 32) * 5 )/ 9;
+    
+    return celcius;
+}
+
+void alertInCelcius(float farenheit, Environment environment_e) 
+{
+    float celcius = convertFarenheitToCelcius(farenheit);    
+    int returnCode = SelectAlertEnvironment(celcius,environment_e);    

```

Comment:
What if accidentally the test code gets executed in the field ? this might lead to unwanted behavior of software.